src/xml2js.coffee

```diff
@@ -255,6 +257,8 @@ class exports.Parser extends events.EventEmitter
       obj = {}
       obj[charkey] = ""
       unless @options.ignoreAttrs
+        if @options.explicitAttrs and attrkey not of obj
```

Comment:
I can't see that it makes any difference to performance, but I moved it anyway, maybe I'm misunderstanding you? It would be neater to avoid two tests, but then you need to check Object.keys(obj).length which might not be available so then the code just becomes longer.